tests/storages_tests/rdb_tests/test_storage.py

```diff
@@ -123,50 +129,62 @@ def test_set_default_engine_kwargs_for_mysql_with_other_rdb() -> None:
 
 
 def test_check_table_schema_compatibility() -> None:
-    storage = create_test_storage()
-    session = storage.scoped_session()
+    with create_test_storage() as storage:
+        session = storage.scoped_session()
 
-    # The schema version of a newly created storage is always up-to-date.
-    storage._version_manager.check_table_schema_compatibility()
-
-    # `SCHEMA_VERSION` has not been used for compatibility check since alembic was introduced.
-    version_info = session.query(VersionInfoModel).one()
-    version_info.schema_version = SCHEMA_VERSION - 1
-    session.commit()
+        # The schema version of a newly created storage is always up-to-date.
+        storage._version_manager.check_table_schema_compatibility()
 
-    storage._version_manager.check_table_schema_compatibility()
+        # `SCHEMA_VERSION` has not been used for compatibility check since alembic was introduced.
+        version_info = session.query(VersionInfoModel).one()
+        version_info.schema_version = SCHEMA_VERSION - 1
+        session.commit()
 
-    with pytest.raises(RuntimeError):
-        storage._version_manager._set_alembic_revision(
-            storage._version_manager._get_base_version()
-        )
         storage._version_manager.check_table_schema_compatibility()
 
-
-def create_test_storage(engine_kwargs: dict[str, Any] | None = None) -> RDBStorage:
-    storage = RDBStorage("sqlite:///:memory:", engine_kwargs=engine_kwargs)
-    return storage
+        with pytest.raises(RuntimeError):
+            storage._version_manager._set_alembic_revision(
+                storage._version_manager._get_base_version()
+            )
+            storage._version_manager.check_table_schema_compatibility()
+
+
+@contextmanager
+def create_test_storage(
+    url: str = "sqlite:///:memory:",
+    engine_kwargs: dict[str, Any] | None = None,
+    skip_compatibility_check: bool = False,
+    skip_table_creation: bool = False,
+) -> Generator[RDBStorage, None, None]:
+    storage = RDBStorage(
+        url,
+        engine_kwargs=engine_kwargs,
+        skip_compatibility_check=skip_compatibility_check,
+        skip_table_creation=skip_table_creation,
+    )
+    yield storage
+    storage.engine.dispose()
```

Comment:
```suggestion
    try:
        yield storage
    finally:
        storage.engine.dispose()
```
it would be better to add `try` and `finally` for readability. (ref: https://github.com/optuna/optuna/pull/6330#discussion_r2521966118)